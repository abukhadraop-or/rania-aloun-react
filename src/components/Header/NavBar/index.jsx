import React, { useEffect, useState } from 'react';
import NavBar from './nav-bar';
import {
  Element,
  MenuElement,
  MenuButton,
  CloseButton,
  CloseElement,
  StyledNavLink,
} from './nav-element';

/**
 * Responsive component for the navigation bar in the header of the page.
 */
function Nav() {
  const [clicked, setClicked] = useState(true);

  /**
   * flips the clicked state in order to show the navbar or hide it
   */
  const handleClick = () => {
    setClicked(!clicked);
  };

  useEffect(() => {
    if (window.screen.width <= 375) {
      setClicked(false);
    }
  }, []);

  return (
    <>
      {/**
       * Menu icon only shown in mobile & if not clicked
       */}
      {!clicked && (
        <MenuButton onClick={handleClick}>
          <MenuElement />
        </MenuButton>
      )}

      {/**
       * Navbar shown in mobile only if menu icon is clicked
       */}
      {clicked && (
        <NavBar>
          <CloseButton onClick={handleClick}>
            <CloseElement />
          </CloseButton>
          <Element>
            <StyledNavLink to="/home"> Home </StyledNavLink>
          </Element>
          <Element>
            <StyledNavLink to="/signIn">Sign in</StyledNavLink>
          </Element>
          <Element>
            <StyledNavLink to="/signUp">Sign up</StyledNavLink>
          </Element>
        </NavBar>
      )}
    </>
  );
}

export default Nav;
