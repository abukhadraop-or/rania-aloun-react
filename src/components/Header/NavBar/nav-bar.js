import styled from "styled-components";

export default styled.ul`
  font-size: 15px;
  background-color: white;
  color: #98a69b;
  list-style-type: none;
  margin: 0;
  padding: 0;
  display: flex;
  align-items: center;
  @media (max-width: 375px) {
    position: absolute;
    top: 0.6rem;
    box-shadow: 0 0 0.2rem rgba(0, 0, 0, 0.8);
    width: fit-content;
    flex-direction: column;
    right: 2.2rem;
  }
`;
