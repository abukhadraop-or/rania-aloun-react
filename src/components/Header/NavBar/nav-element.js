import styled from 'styled-components';
import { ReactComponent as menu } from 'assets/menu.svg';
import { ReactComponent as close } from 'assets/close.svg';
import { NavLink } from 'react-router-dom';

export const Element = styled.li`
  display: inline;
  margin: 0 10px 0;
  cursor: pointer;
  &:hover {
    color: #616d72;
  }
  @media (max-width: 375px) {
    display: flex;
    flex-direction: column;
    margin: 0 0.5rem 1rem;
  }
`;

export const CloseButton = styled.button`
  display: none;
  border: none;
  cursor: pointer;
  background-color: transparent;
  margin: 0 0.5rem 1.5rem;
  @media (max-width: 375px) {
    display: block;
  }
`;

export const CloseElement = styled(close)`
  display: none;
  margin: 0.4rem 0;
  @media (max-width: 375px) {
    display: block;
    align-items: center;
  }
`;

export const MenuButton = styled.button`
  display: none;
  border: none;
  cursor: pointer;
  background-color: transparent;
  @media (max-width: 375px) {
    display: block;
  }
`;

export const MenuElement = styled(menu)``;

export const StyledNavLink = styled(NavLink)`
  text-decoration: none;
  color: inherit;
`;
