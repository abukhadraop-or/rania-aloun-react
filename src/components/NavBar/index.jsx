import React, { useState } from 'react';
import {
  MenuIcon,
  Container,
  CloseIcon,
  NavElement,
  MenuButton,
  StyledNavLink,
} from './nav-bar.styles';

/**
 * Responsive component holding the navigation bar in the header of the page.
 *
 * @return {JSX.Element} Navigation bar holding home, sign in, and sign up routes.
 */
function NavBar() {
  const [isClicked, setClicked] = useState(false);

  const handleClick = () => {
    setClicked(!isClicked);
  };

  return (
    <>
      {isClicked && (
        <MenuButton onClick={handleClick}>
          <CloseIcon />
        </MenuButton>
      )}
      <Container isClicked={isClicked}>
        <NavElement>
          <StyledNavLink to="/home"> Home </StyledNavLink>
        </NavElement>
        <NavElement>
          <StyledNavLink to="/sign-in">Sign in</StyledNavLink>
        </NavElement>
        <NavElement>
          <StyledNavLink to="/sign-up">Sign up</StyledNavLink>
        </NavElement>
      </Container>
      {!isClicked && (
        <MenuButton onClick={handleClick}>
          <MenuIcon />
        </MenuButton>
      )}
    </>
  );
}

export default NavBar;
