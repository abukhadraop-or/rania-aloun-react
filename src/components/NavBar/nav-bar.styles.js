import styled from 'styled-components';
import { ReactComponent as menu } from 'assets/menu.svg';
import { ReactComponent as close } from 'assets/close.svg';
import { NavLink } from 'react-router-dom';
import colors from 'styles/colors';
import breakpoints from 'styles/break-points';

export const Container = styled.ul`
  align-items: center;
  background-color: white;
  color: ${colors.coolGrey};
  display: flex;
  font-size: 0.9375rem;
  list-style-type: none;
  margin: 0;

  @media (max-width: ${breakpoints.mobile}) {
    box-shadow: 0 3px 2px -2px rgba(0, 0, 0, 0.8);
    flex-direction: column;
    display: ${({ isClicked }) => (isClicked ? 'static' : 'none')};
    position: absolute;
    top: 3.5rem;
    width: 100%;
    right: 0;
    padding: 0;
  }
`;

export const NavElement = styled.li`
  cursor: pointer;
  display: inline;
  margin: 0 10px 0;

  &:hover {
    color: ${colors.shuttleGrey};
  }

  @media (max-width: ${breakpoints.mobile}) {
    border-top: 1px solid ${colors.ashGrey};
    display: flex;
    flex-direction: column;
    padding: 0.5rem 0;
    right: 0;
    width: 100%;
  }
`;

export const CloseIcon = styled(close)`
  display: none;
  margin: 0.4rem 0;

  @media (max-width: ${breakpoints.mobile}) {
    align-items: center;
    display: block;
  }
`;

export const MenuButton = styled.button`
  background-color: transparent;
  border: none;
  cursor: pointer;
  display: none;

  @media (max-width: ${breakpoints.mobile}) {
    display: block;
  }
`;

export const MenuIcon = styled(menu)``;

export const StyledNavLink = styled(NavLink)`
  color: inherit;
  text-decoration: none;
`;
