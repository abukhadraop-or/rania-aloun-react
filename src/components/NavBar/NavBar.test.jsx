import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import NavBar from '.';

describe('<NavBar />', () => {
  it('matches snapshot', () => {
    const wrapper = shallow(<NavBar />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
