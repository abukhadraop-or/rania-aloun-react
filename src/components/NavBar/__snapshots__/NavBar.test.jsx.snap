// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavBar /> matches snapshot 1`] = `
<Fragment>
  <styled.ul
    isClicked={false}
  >
    <styled.li>
      <Styled(NavLink)
        to="/home"
      >
         Home 
      </Styled(NavLink)>
    </styled.li>
    <styled.li>
      <Styled(NavLink)
        to="/sign-in"
      >
        Sign in
      </Styled(NavLink)>
    </styled.li>
    <styled.li>
      <Styled(NavLink)
        to="/sign-up"
      >
        Sign up
      </Styled(NavLink)>
    </styled.li>
  </styled.ul>
  <styled.button
    onClick={[Function]}
  >
    <Styled(Component) />
  </styled.button>
</Fragment>
`;
